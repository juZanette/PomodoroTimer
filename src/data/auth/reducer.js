const initialState = {
  signed: false,
}

export const reducer = (state = initialState, action) => {
  switch (action.type) {
    case 'SIGN_IN':
      return {
        signed: true,
      }
    default:
      return state
  }
}
