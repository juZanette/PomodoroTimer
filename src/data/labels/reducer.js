const initialState = {
  dialogOpened: false,
  labelToEdit: null,
  formValue: {
    name: null,
    color: null,
  },
}

export const reducer = (state = initialState, action) => {
  switch (action.type) {
    case 'SET_DIALOG_OPENED':
      return {
        ...state,
        dialogOpened: action.dialogOpened,
      }
    case 'SET_LABEL_TO_EDIT':
      return {
        ...state,
        labelToEdit: action.labelToEdit,
      }
    case 'SET_FORM_VALUE':
      return {
        ...state,
        formValue: action.formValue,
      }
    default:
      return state
  }
}
