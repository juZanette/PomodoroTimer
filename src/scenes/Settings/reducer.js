const initialState = {
  rounds: 4,
}

export const reducer = (state = initialState, action) => {
  switch (action.type) {
    case 'SET_ROUNDS':
      return {
        rounds: action.roundsNumber,
      }
    default:
      return state
  }
}
