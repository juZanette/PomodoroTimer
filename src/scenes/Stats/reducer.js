const initialState = {
  totalSessions: 0,
}

export const reducer = (state = initialState, action) => {
  switch (action.type) {
    case 'INCREMENT_TOTAL_SESSIONS':
      return {
        totalSessions: ++state.totalSessions,
      }
    default:
      return state
  }
}
