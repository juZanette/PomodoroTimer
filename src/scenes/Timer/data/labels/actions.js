export const setDialogOpened = (dialogOpened) => ({
  type: 'SET_DIALOG_OPENED',
  dialogOpened,
})

export const setAlertOpened = (alertOpened) => ({
  type: 'SET_ALERT_OPENED',
  alertOpened,
})

export const setLabelToEdit = (labelToEdit) => ({
  type: 'SET_LABEL_TO_EDIT',
  labelToEdit,
})

export const setFormValue = (formValue) => ({
  type: 'SET_FORM_VALUE',
  formValue,
})
