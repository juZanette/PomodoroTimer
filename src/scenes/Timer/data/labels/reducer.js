const initialState = {
  dialogOpened: false,
  formDialogOpened: false,
  deleteAlert: {
    opened: false,
    labelToDelete: null,
  },
  labelToEdit: null,
  formValue: null,
}

export const reducer = (state = initialState, action) => {
  switch (action.type) {
    case 'SET_DIALOG_OPENED':
      return {
        ...state,
        dialogOpened: action.dialogOpened,
      }
    case 'SET_FORM_DIALOG':
      return {
        ...state,
        formDialogOpened: action.formDialogOpened,
      }
    case 'SET_DELETE_ALERT':
      const { opened, labelToDelete } = action.deleteAlert

      return {
        ...state,
        deleteAlert: {
          opened,
          labelToDelete,
        },
      }
    case 'SET_LABEL_TO_EDIT':
      return {
        ...state,
        labelToEdit: action.labelToEdit,
      }
    case 'SET_FORM_VALUE':
      return {
        ...state,
        formValue: action.formValue,
      }
    default:
      return state
  }
}
