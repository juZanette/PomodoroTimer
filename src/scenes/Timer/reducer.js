const initialState = {
  timerActive: false,
}

export const reducer = (state = initialState, action) => {
  switch (action.type) {
    case 'SET_TIMER_ACTIVE':
      return {
        timerActive: action.timerActive,
      }
    default:
      return state
  }
}
