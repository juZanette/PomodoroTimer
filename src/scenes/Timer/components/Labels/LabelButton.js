import React, { useRef } from 'react'
import { useSelector, useDispatch } from 'react-redux'
import styled from 'styled-components'
import { useTheme, useMediaQuery } from '@material-ui/core'
import Box from '@material-ui/core/Box'
import MatButton from '@material-ui/core/Button'
import LabelIcon from '@material-ui/icons/Label'
import {
  setFullscreenDialog,
  setDesktopDialog,
  setMenuOpened,
} from '../../../../data/labels/actions'
import { LabelsMenu } from './LabelsMenu'
import { STATUSES } from '../../data/timer/reducer'

export const LabelButton = () => {
  const { labelSelected, data } = useSelector((state) => state.labels)
  const { status } = useSelector((state) => state.timer)

  const dispatch = useDispatch()

  const buttonRef = useRef(null)

  const theme = useTheme()
  const isMediumScreen = useMediaQuery(theme.breakpoints.up('md'))

  const openDialog = () => {
    if (isMediumScreen) {
      dispatch(setDesktopDialog(true))
    } else {
      dispatch(setFullscreenDialog(true))
    }
  }

  const handleClick = () => {
    if (data && data.length) {
      dispatch(setMenuOpened(true))
    } else {
      openDialog()
    }
  }

  const getButtonText = () => {
    if (labelSelected) {
      return labelSelected.name
    }

    if (data && data.length) {
      return 'Select label'
    }

    return 'Add label'
  }

  return (
    <Box display="flex" justifyContent="center">
      <Button
        disabled={status !== STATUSES.onHold}
        labelcolor={
          labelSelected ? labelSelected.color : theme.palette.secondary.main
        }
        size="small"
        startIcon={<LabelIcon />}
        onClick={handleClick}
        ref={buttonRef}
      >
        {getButtonText()}
      </Button>

      <LabelsMenu anchor={buttonRef.current} />
    </Box>
  )
}

export const Button = styled(MatButton)`
  color: ${({ labelcolor }) => labelcolor};
`
