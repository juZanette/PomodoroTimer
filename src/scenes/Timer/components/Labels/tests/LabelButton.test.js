import React from 'react'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { createMount } from '@material-ui/core/test-utils'
import labels from './fixtures/labels'
import { LabelButton, Button } from '../LabelButton'
import { STATUSES } from '../../../data/timer/reducer'

describe('<LabelButton />', () => {
  const mockStore = configureMockStore([thunk])

  let store
  let mount
  let wrapper

  beforeEach(() => {
    mount = createMount()

    wrapper = (
      labelsData = [],
      labelSelected = null,
      timerStatus = STATUSES.onHold
    ) => {
      const storeData = {
        labels: { data: labelsData, labelSelected },
        timer: { status: timerStatus },
      }

      store = mockStore(storeData)
      store.dispatch = jest.fn()

      return mount(
        <Provider store={store}>
          <LabelButton />
        </Provider>
      )
    }
  })

  test('should render <LabelButton /> correctly', () => {
    expect(wrapper()).toMatchSnapshot()
  })

  test('button should be enabled if timer status is ON_HOLD', () => {
    expect(wrapper().find(Button).prop('disabled')).toBe(false)
  })

  test('button should be disabled if timer status is not ON_HOLD', () => {
    expect(
      wrapper([], null, STATUSES.running).find(Button).prop('disabled')
    ).toBe(true)
  })

  test('button should have text ADD LABEL if there are no labels created', () => {
    expect(wrapper().find(Button).text().toUpperCase()).toBe('ADD LABEL')
  })

  test('button should have text SELECT LABEL if there are some labels created', () => {
    expect(wrapper(labels).find(Button).text().toUpperCase()).toBe(
      'SELECT LABEL'
    )
  })

  test('button should display the text of the selected label if a label is selected', () => {
    expect(wrapper(labels, labels[0]).find(Button).text()).toBe(labels[0].name)
  })

  test('should call dispatch once on button click if there are no labels created', () => {
    wrapper().find(Button).simulate('click')
    expect(store.dispatch).toHaveBeenCalledTimes(1)
  })

  test('should call dispatch once on button click if there are some labels created', () => {
    wrapper(labels).find(Button).simulate('click')
    expect(store.dispatch).toHaveBeenCalledTimes(1)
  })
})
