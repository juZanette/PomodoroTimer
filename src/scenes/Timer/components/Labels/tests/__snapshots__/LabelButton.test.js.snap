// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LabelButton /> should render <LabelButton /> correctly 1`] = `
<Styled(MuiBox)
  display="flex"
  justifyContent="center"
>
  <Styled(WithStyles(ForwardRef(Button)))
    disabled={false}
    labelcolor="#f50057"
    onClick={[Function]}
    size="small"
    startIcon={<UNDEFINED />}
  >
    Add label
  </Styled(WithStyles(ForwardRef(Button)))>
  <LabelsMenu
    anchor={null}
  />
</Styled(MuiBox)>
`;
